### ISSUE TITLE
Why can't the model fit normally when the data set is expanded?

### ISSUE THREAD

**Issue by jetyang0729** [ultralytics/yolov5#7222]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

First of all, I want to thank you for opening up such an excellent model. This model is very shocked for beginners like me. I really appreciate your excellent work.

When I first tested the model, I used about 198 pictures, 1600 labeled objects and yolov5x models. The training was set to 300 epochs, but stopped at about 200 because they didn't get a better weight within 100 epochs. I got a very good best Pt, and can well adapt to my data.

So I increased my data set to 8000 pieces. As a result, when I trained 190 times (for some reasons, I divided the 190 epochs into three times), I told me that the 100 epochs could not get a better best pt。 Then after my test, I found that this large data set did not let me get a better model, whether using the pictures in the training set or the pictures in the verification set. The second model fitting is far worse than the first model, and even the second model cannot recognize the validation set.

Based on the above, I would like to ask some questions about models and data sets:
1. Why can't the second data set find a better model? Because in the result Figure 1, I can find that loss is not optimal. At the same time, I wonder why train appears_ When loss decreases, Val_ Is loss rising instead? And the values of P and Q also decrease? What is the standard to judge best PT is better than last PT better?
2. About the dataset, you wrote in the dataset suggestion: all instances of all classes in all images must be labeled Partial labelling will not work.  What will happen if some required targets are missed? Secondly, if there are many targets in this picture, but I only need to identify individual targets, do I need to mark other targets? Is there any problem if it is not marked? Finally, it is about the background picture: how can it be regarded as a qualified background picture? No goals at all? Or as long as there are no marked targets? Where is the meaning of the background picture that can be withdrawn? Or is there any evidence to prove that the background picture has a very important meaning in the field of detection?
![result1](https://user-images.githubusercontent.com/102574953/161021289-2bfcd9a3-6171-4005-a0ce-cf697ddced43.jpg)
![results](https://user-images.githubusercontent.com/102574953/161021299-f67e8e03-b461-44c8-a8fa-d4ec53718eda.png)
![result6](https://user-images.githubusercontent.com/102574953/161021300-91b1cff9-3945-4f4d-9c8d-689c23cc6166.jpg)
![result2](https://user-images.githubusercontent.com/102574953/161021304-648aed09-a7ae-4d3f-89c7-03f6f083d443.jpg)
![result3](https://user-images.githubusercontent.com/102574953/161021307-cf1a6ac2-dddb-4b7a-a414-b94f7134f9e6.jpg)
![result4](https://user-images.githubusercontent.com/102574953/161021309-f62e46b7-b2dd-4d4c-af12-2d8cdd63044d.jpg)
![result5](https://user-images.githubusercontent.com/102574953/161021311-f2f55ad4-a2dc-4a62-a9ea-982a073e997e.jpg)

**Comment by glenn-jocher** [comment]
@jetyang0729 additional data will improve results if the quality/domain etc is similar to the existing data. For example training with 20% of the COCO dataset will produce better results than training with 10% COCO dataset. If you add data and your results worsen, then it's likely there is a problem with your additional data, or perhaps you've introduced additional classes and your data per class has actually dropped etc.

All instances of objects must be labelled in all your images. If an object is not labelled the model will actively be taught that there is no object there.

Background images are not required but can be used to reduce False Positives in real world deployments. i.e. if you see FPs in a deployed environment then add background images from the deployed environment without any labels to teach the model that there are no objects there.

**Comment by glenn-jocher** [comment]
@jetyang0729 also see our full guide for improving results below.

Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥ 1500 images per class recommended
- **Instances per class.** ≥ 10000 instances (labeled objects) per class recommended
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total). No labels are required for background images.

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. [VOC](https://github.com/ultralytics/yolov5/blob/master/data/VOC.yaml), [VisDrone](https://github.com/ultralytics/yolov5/blob/master/data/VisDrone.yaml), [GlobalWheat](https://github.com/ultralytics/yolov5/blob/master/data/GlobalWheat2020.yaml)). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
                                             custom_pretrained.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. [COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [Objects365](https://github.com/ultralytics/yolov5/blob/master/data/Objects365.yaml), [OIv6](https://storage.googleapis.com/openimages/web/index.html)). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyp.scratch.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains: http://karpathy.github.io/2019/04/25/recipe/

Good luck 🍀 and let us know if you have any other questions!

**Comment by jetyang0729** [comment]
> @jetyang0729 additional data will improve results if the quality/domain etc is similar to the existing data. For example training with 20% of the COCO dataset will produce better results than training with 10% COCO dataset. If you add data and your results worsen, then it's likely there is a problem with your additional data, or perhaps you've introduced additional classes and your data per class has actually dropped etc.
> 
> All instances of objects must be labelled in all your images. If an object is not labelled the model will actively be taught that there is no object there.
> 
> Background images are not required but can be used to reduce False Positives in real world deployments. i.e. if you see FPs in a deployed environment then add background images from the deployed environment without any labels to teach the model that there are no objects there.

Thank you very much for your reply to the data set question. I agree with you very much. However, for some reasons, the possibility that I can reach the ideal data set is very low, so I hope to make the model adapt well to the model in the case of poor data sets. So I hope to know:
1. What is the reason why the loss in my results is not optimal, but its trend is declining (I think this trend is the process of the model getting better!). But in this case, why does train_loss decrease and Val_loss increase? And the values of P and Q also decrease?
2. What criteria are used to determine the best.pt is the best weight? P and Q? Or Val_ loss?

**Comment by glenn-jocher** [comment]
@jetyang0729 your model is overfitting after the first few epochs. 

best.pt is saved on every best fitness epoch with fitness defined as a weighted combination of metrics:
https://github.com/ultralytics/yolov5/blob/71621df87589faea19ba4c4098bb68e73201f30c/utils/metrics.py#L15-L19

**Comment by jetyang0729** [comment]
> @jetyang0729 your model is overfitting after the first few epochs.
> 
> best.pt is saved on every best fitness epoch with fitness defined as a weighted combination of metrics:
> 
> https://github.com/ultralytics/yolov5/blob/71621df87589faea19ba4c4098bb68e73201f30c/utils/metrics.py#L15-L19
@glenn-jocher Thank you again for your reply! This is very important for my research!
I've started retraining and reduced Lr0 and LR1 in hyp (to reduce the learning rate). Can this solve the problem of over fitting the results? Or can you give me some suggestions to solve over fitting?
Thanks a lot!

**Comment by glenn-jocher** [comment]
@jetyang0729 to reduce overfitting you can try using a hyp.scratch-high.yaml, gather more data, train less epochs, train from scratch, etc.

**Comment by jetyang0729** [comment]
> @jetyang0729 to reduce overfitting you can try using a hyp.scratch-high.yaml, gather more data, train less epochs, train from scratch, etc.
@glenn-jocher thanks a lot for your advice. Actually.I developoed a new yaml with lower learning rate and solved the problem of over fitting. However,I still cannot get a model to deteted the val images. I think the main problem is my dataset is pretty bad. I will re-label and collect my data to achieve a balance of instances for each class.
I also want to ask another question: because my data collection is very difficult and sensitive, I hope to improve these situations in other ways. I don't know if you know 'copy – move attack detection'. If I directly copy instances with fewer classes in my dataset to supplement this identification process. Can this approach work?

**Comment by glenn-jocher** [comment]
@jetyang0729 you can try `python trian.py --image-weights`, this setting samples images according to their performance to overrepresent poorly performing classes, which may help. It's highly experimental though, so results will vary:
https://github.com/ultralytics/yolov5/blob/37675e110f3d2635dbc3acc8794e782c452e4ad5/train.py#L487
